### PR TITLE
Require class definition in metadata constructors

### PR DESCRIPTION
**Pull request by j9liu**
This PR makes a `MetadataClass` definition a requirement in `MetadataEntity` and consequently, all other metadata-related classes. It deletes the unit tests that expected functionality without a class and updates the others so they don't fail.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Specs/Scene/MetadataTableSpec.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ describe("Scene/MetadataTable", function () {
     );
   });
 
-  it("constructor throws without count", function () {
+  it("constructor throws without parameters", function () {
```
Comment (ptrgags) [inline Specs/Scene/MetadataTableSpec.js:67]: I think we do want to keep this as "throws without count", just pass in `{}` for class

**Review by ptrgags — COMMENTED** [review]
@j9liu This looks most of the way there, just a few changes needed.

File: Source/Scene/JsonMetadataTable.js [hunk 2]
```diff
@@ -15,6 +15,9 @@ import MetadataEntity from "./MetadataEntity.js";
  * @constructor
  * @private
  */
+
+const emptyClass = {};
```
Comment (ptrgags) [inline Source/Scene/JsonMetadataTable.js:19]: Maybe put a comment here along the lines of "JsonMetadataTable is an older type of metadata table that does not have a class definition"

**Comment by ptrgags** [comment]
Looks good, thanks @j9liu!

**Pull request merged**